misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java

```diff
@@ -80,7 +88,15 @@
                 }
 
                 String parameterName = params.get(i).getName().get();
-                argValues.put(parameterName, arg.getValue().evaluate().getJdiValue());
+                Value argValue = arg.getValue().evaluate().getJdiValue();
+                // For the ballerina function parameters with type "any", the generated runtime method will only accept
+                // the args which are subtypes of "java.lang.Object". Therefore all the primitive typed arguments
+                // must be converted into their wrapper implementations (i.e. 'int' -> 'Integer'), before passing
+                // into the method.
+                if (params.get(i).typeDescriptor().typeKind().getName().equals(BVariableType.ANY.getString())) {
+                    argValue = getValueAsObject(context, argValue);
+                }
+                argValues.put(parameterName, argValue);
```

Comment:
Shall we add this to a util method and use it here?

Reply:
This is already a util method (`getValueAsObject(context, argValue)`);